src/test/java/org/opensearch/jobscheduler/JobSchedulerPluginTests.java

```diff
@@ -0,0 +1,169 @@
+import org.opensearch.common.settings.IndexScopedSettings;
+import org.opensearch.common.settings.SettingsFilter;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.apache.lucene.tests.index.AssertingDirectoryReader;
+import org.junit.Before;
+import org.mockito.Mock;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.index.Index;
+import org.opensearch.env.Environment;
+import org.opensearch.index.IndexModule;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.index.analysis.AnalysisRegistry;
+import org.opensearch.index.engine.EngineConfigFactory;
+import org.opensearch.jobscheduler.rest.action.RestGetJobDetailsAction;
+import org.opensearch.jobscheduler.rest.action.RestGetLockAction;
+import org.opensearch.jobscheduler.rest.action.RestReleaseLockAction;
+import org.opensearch.jobscheduler.spi.JobSchedulerExtension;
+import org.opensearch.jobscheduler.spi.ScheduledJobParser;
+import org.opensearch.jobscheduler.spi.ScheduledJobRunner;
+import org.opensearch.jobscheduler.utils.JobDetailsService;
+import org.opensearch.plugins.ExtensiblePlugin;
+import org.opensearch.rest.RestController;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.test.IndexSettingsModule;
+import org.opensearch.test.OpenSearchTestCase;
+
+import org.opensearch.test.engine.MockEngineFactory;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class JobSchedulerPluginTests extends OpenSearchTestCase {
+
+    private JobSchedulerPlugin plugin;
+
+    private Index index;
+    private Settings settings;
+    private Settings sweeperSettings;
+    private IndexSettings indexSettings;
+    private IndexModule indexModule;
+    private AnalysisRegistry emptyAnalysisRegistry;
+
+    @Mock
+    private RestController restController;
+    @Mock
+    private ClusterSettings clusterSettings;
+    @Mock
+    private IndexScopedSettings indexScopedSettings;
+    @Mock
+    private SettingsFilter settingsFilter;
+    @Mock
+    private IndexNameExpressionResolver indexNameExpressionResolver;
+    @Mock
+    private Supplier<DiscoveryNodes> nodesInCluster;
+
+    @Before
+    public void setup() {
+        plugin = new JobSchedulerPlugin();
+        settings = Settings.builder()
+            .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+            .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
+            .build();
+        indexSettings = IndexSettingsModule.newIndexSettings(".opensearch-job-scheduler-job-details", settings);
```

Comment:
Import this from JobDetailsService?

Reply:
Let me check this.

Reply:
Just pushed in my latest commit.
Thanks